modin/engines/base/frame/data.py

```diff
@@ -220,11 +220,20 @@ def _set_axis(self, axis, new_axis):
                 Axis to set labels along
             new_axis : Index,
                 The replacement labels
+            cache_only : bool,
+                Whether to change only external indices, or propagate it
+                into partitions
         """
         if axis:
-            self._set_columns(new_axis)
+            if not cache_only:
+                self._set_columns(new_axis)
+            else:
+                self._index_cache = ensure_index(new_axis)
```

Comment:
Why `self._index_cache` for `axis == 1`?

Reply:
typo, thanks for notice